# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/webrtc/signaling/src/peerconnection/PeerConnectionImpl.cpp
# Commit: 0b03dd9d20ac
# Full Hash: 0b03dd9d20ace235502714101097fe34d1864563
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-07-31 15:46:43
# Regressor Bug: 1493613
# File Overlap Count: 1
# Description:
#   Bug 1493613 - Update muted state through MediaStreamTrackSource. r=bwc,smaug
#   
#   This ensures all clones of the original track also receives the new muted state.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D37933
# ==============================================================================

diff -r cf84eeffed22 -r 0b03dd9d20ac media/webrtc/signaling/src/peerconnection/PeerConnectionImpl.cpp
--- a/media/webrtc/signaling/src/peerconnection/PeerConnectionImpl.cpp	Tue Jul 30 14:55:48 2019 +0000
+++ b/media/webrtc/signaling/src/peerconnection/PeerConnectionImpl.cpp	Wed Jul 31 09:35:06 2019 +0000
@@ -1824,25 +1824,24 @@
   }
 
   RefPtr<MediaStreamTrack> track;
+  RefPtr<RemoteTrackSource> trackSource;
   if (audio) {
-    track = stream->CreateDOMTrack(
-        333,  // Use a constant TrackID. Dependents read this from the DOM
-              // track.
-        MediaSegment::AUDIO,
-        new RemoteTrackSource(principal,
-                              NS_ConvertASCIItoUTF16("remote audio")));
+    trackSource = new RemoteTrackSource(principal,
+                                        NS_ConvertASCIItoUTF16("remote audio"));
+    track = stream->CreateDOMTrack(333,  // Use a constant TrackID. Dependents
+                                         // read this from the DOM track.
+                                   MediaSegment::AUDIO, trackSource);
   } else {
-    track = stream->CreateDOMTrack(
-        666,  // Use a constant TrackID. Dependents read this from the DOM
-              // track.
-        MediaSegment::VIDEO,
-        new RemoteTrackSource(principal,
-                              NS_ConvertASCIItoUTF16("remote video")));
+    trackSource = new RemoteTrackSource(principal,
+                                        NS_ConvertASCIItoUTF16("remote video"));
+    track = stream->CreateDOMTrack(666,  // Use a constant TrackID. Dependents
+                                         // read this from the DOM track.
+                                   MediaSegment::VIDEO, trackSource);
   }
 
   stream->AddTrackInternal(track);
   // Spec says remote tracks start out muted.
-  track->MutedChanged(true);
+  trackSource->SetMuted(true);
 
   return OwningNonNull<dom::MediaStreamTrack>(*track);
 }